include/soci/odbc/soci-odbc.h

```diff
@@ -545,6 +548,25 @@ inline void odbc_standard_type_backend_base::set_sqllen_from_value(SQLLEN &targe
     }
 }
 
+inline bool odbc_standard_type_backend_base::supports_negative_tinyint() const
+{
+    // MSSQL ODBC driver only supports a range of [0..255] for tinyint.
+    return statement_.session_.get_database_product()
+            != odbc_session_backend::prod_mssql;
+}
+
+inline bool odbc_standard_type_backend_base::can_convert_to_unsigned_sql_type() const
+{
+    // MSSQL ODBC driver seemingly can't handle the conversion of unsigned C
+    // types to their respective unsigned SQL type because they are out of
+    // range for their supported signed types. This results in the error
+    // "Numeric value out of range (SQL state 22003)".
+    // The only place it works is with tinyint values as their range is
+    // [0..255], i.e. they have enough space for unsigned values anyway.
+    return statement_.session_.get_database_product()
+            != odbc_session_backend::prod_mssql;
+}
+
```

Comment:
All of these helper functions are declared and defined in the header, so I just continued this convention.

> I suspect we might need to change/adjust them in the future to account for more ODBC quirks (e.g. test the exact version or something else).

Because of this I would just leave them as is if that's okay with you.